src/algorithm/verifyDpopProof.ts

```diff
@@ -50,7 +50,7 @@ export async function verifyDpopProof(
   const dpop = {
     header: protectedHeader,
     payload,
-    signature: dpopHeader.split(".")[2],
+    signature: /(?<=^\w+\.\w+\.)\w*/.exec(dpopHeader)![0],
```

Comment:
What could possibly make that fail?

I find the replacement regex a little hard to read and I'm not sure it achieves more.

Reply:
Same as my other comment, I think sticking to a simple split for extracting access token segments would be better.

Reply:
Yeah agree, I forgot to mention in my PR text that I just replaced all the `splits` and that you could mix and match, but that was overzealous.